opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+        has_gains=False,
+        max_gains=ControlGains(kp=1000, ki=1000, kd=1000, k=0, b=0, ff=0),
+    ),
+    CURRENT=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_current_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=0, b=0, ff=128),
+    ),
+    VOLTAGE=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_voltage_mode_exit,
+        has_gains=False,
+        max_gains=None,
+    ),
+    IMPEDANCE=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_impedance_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=1000, b=1000, ff=128),
+    ),
+)
+
+
+class DephyActuator(ActuatorBase, Device):
+    def __init__(
+        self,
+        tag: str = "DephyActuator",
+        firmware_version: str = "7.2.0",
+        port: str = "/dev/ttyACM0",
+        gear_ratio: float = 1.0,
+        baud_rate: int = 230400,
+        frequency: int = 500,
+        debug_level: int = 0,
+        dephy_log: bool = False,
+        offline: bool = False,
+        stop_motor_on_disconnect: bool = False,
+    ) -> None:
+        ActuatorBase.__init__(
+            self,
+            tag=tag,
+            gear_ratio=gear_ratio,
+            motor_constants=DEPHY_ACTUATOR_CONSTANTS,
+            frequency=frequency,
+            offline=offline,
+        )
+
+        if self.is_offline:
+            self.port = port
+        else:
+            Device.__init__(
+                self,
+                firmwareVersion=firmware_version,
+                port=port,
+                baudRate=baud_rate,
+                stopMotorOnDisconnect=stop_motor_on_disconnect,
+            )
+
+        self._debug_level: int = debug_level
+        self._dephy_log: bool = dephy_log
+
+        self._thermal_model: ThermalModel = ThermalModel(
+            temp_limit_windings=self.max_winding_temperature,
+            soft_border_C_windings=10,
+            temp_limit_case=self.max_case_temperature,
+            soft_border_C_case=10,
+        )
+        self._thermal_scale: float = 1.0
+
+        self._mode = CONTROL_MODES.VOLTAGE
+
+    def __repr__(self) -> str:
+        return f"{self.tag}[DephyActuator]"
+
+    @property
+    def _CONTROL_MODE_CONFIGS(self) -> CONTROL_MODE_CONFIGS:
+        return DEPHY_CONTROL_MODE_CONFIGS
+
+    @check_actuator_connection
+    def start(self) -> None:
+        try:
+            self.open()
+        except OSError as e:
+            print("\n")
+            LOGGER.error(
+                msg=f"[{self.__repr__()}] Need admin previleges to open the port '{self.port}'. \n\nPlease run the script with 'sudo' command or add the user to the dialout group.\n"
+            )
+            os._exit(status=1)
+
+        self.start_streaming(self._frequency)
+        self._data = self.read()
+
+        time.sleep(0.1)
```

Comment:
Rouge time.sleep - not sure if you want this here still or not

Reply:
Yes! Marking this as a todo, which can be then added to the issue / project board.